src/status_im/contexts/wallet/wallet_connect/events/session_proposals.cljs

```diff
@@ -119,33 +118,51 @@
                                (map networks/chain-id->eip155)
                                vec)
          current-address  (get-in db [:wallet-connect/current-proposal :address])
-         accounts         (-> (partial networks/format-eip155-address current-address)
-                              (map session-networks))
          network-status   (:network/status db)
-         expiry           (get-in current-proposal [:params :expiryTimestamp])]
+         expired?         (-> current-proposal
+                              (get-in [:params :expiryTimestamp])
+                              uri/timestamp-expired?)]
      (if (= network-status :online)
        {:db (assoc-in db [:wallet-connect/current-proposal :response-sent?] true)
-        :fx [(if (uri/timestamp-expired? expiry)
+        :fx [(if expired?
                [:dispatch
                 [:toasts/upsert
                  {:id   :wallet-connect-proposal-expired
                   :type :negative
                   :text (i18n/label :t/wallet-connect-proposal-expired)}]]
                [:effects.wallet-connect/approve-session
-                {:web3-wallet web3-wallet
-                 :proposal    current-proposal
-                 :networks    session-networks
-                 :accounts    accounts
-                 :on-success  (fn [approved-session]
-                                (log/info "Wallet Connect session approved")
-                                (rf/dispatch [:wallet-connect/reset-current-session-proposal])
-                                (rf/dispatch [:wallet-connect/persist-session
-                                              approved-session]))
-                 :on-fail     (fn [error]
-                                (log/error "Wallet Connect session approval failed"
-                                           {:error error
-                                            :event :wallet-connect/approve-session})
-                                (rf/dispatch
-                                 [:wallet-connect/reset-current-session-proposal]))}])
+                {:web3-wallet      web3-wallet
+                 :proposal-request current-proposal
+                 :session-networks session-networks
+                 :address          current-address
+                 :on-success       #(rf/dispatch [:wallet-connect/approve-session-success %])
+                 :on-fail          #(rf/dispatch [:wallet-connect/approve-session-error %])}])
```

Comment:
Can we use `rf/call-continuation` inside the definition for `:effects.wallet-connect/approve-session`?

If we do that then we can define `:on-success` and `:on-failure` with vectors:
```clojure
[:effects.wallet-connect/approve-session
   {:web3-wallet      web3-wallet
    :proposal-request current-proposal
    :session-networks session-networks
    :address          current-address
    :on-success       [:wallet-connect/approve-session-success]
    :on-fail          [:wallet-connect/approve-session-error]}]
```

For reference, here's an example of how the effect would use `rf/call-continuation` to work promesa: https://github.com/status-im/status-mobile/blob/seanstrom/chore-refactor-screens-definitions/src/status_im/contexts/wallet/effects.cljs#L63-L68